src/wasm-binary.h

```diff
@@ -213,62 +205,46 @@ class BufferWithRandomAccess : public std::vector<uint8_t> {
   }
   BufferWithRandomAccess& operator<<(U32LEB x) {
     size_t before = -1;
-    if (debug) {
-      before = size();
-      std::cerr << "writeU32LEB: " << x.value << " (at " << before << ")"
-                << std::endl;
-    }
+    BYN_DEBUG(before = size(); std::cerr << "writeU32LEB: " << x.value
```

Comment:
It's a little strange that the debug statement has a side effect of assigning to `before` here. It looks like `before` is only used inside the `BYN_DEBUG` code, so I think removing it and just using `size()` would be a good simplification. The same goes for the other `before` variables in subsequent functions.

Reply:
I had a go, but the second debug block wants to use the old size `before`